### PR TITLE
CLOUD-194-TLS updated format to rst, updated index

### PR DESCRIPTION
**Pull request by patrickbirch**
TLS document and added TLS to the index.rst.

### CODE REVIEW

File: source/TLS.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+The tool, cert-manager, automates the management and issuance of TLS
+certificates.
+
+To verify if the certificate is signed and approved, run the following
+command:
+
+.. code:: bash
+
+   kubectl get csr
```
Comment (delgod) [inline source/TLS.rst]: not really :(

File: source/TLS.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
+TLS
```
Comment (delgod) [inline source/TLS.rst]: this doc should describe to the user that he needs to configure internal and external TLS certificates.
explain that it is possible to disable TLS with `allowUnsafeConfigurations` and provide steps on how to generate certificates manually or send user installing cert-manager.

**Comment by delgod** [comment]
also, it is needed to add a step to k8s/openshift install guides about the creation of certificates
```
kubectl apply -f deploy/ssl-secrets.yaml
```

File: source/TLS.rst (hunk 2 above)
Comment (patrickbirch) [reply source/TLS.rst]: Sure. Please provide more information and examples on how someone would configure internal and external TLS certificates, how to disable TLS, and how to generate certificates. 

https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/

How is what we want different than the options in the link? What are the specifically Percona-based methods?

Comment (delgod) [reply source/TLS.rst]: we can have a call if you need more information.
"managing-tls-in-a-cluster" link is **completely** about another thing.

**Comment by patrickbirch** [comment]
Rewrote document based on the discussion.

**Comment by patrickbirch** [comment]
Peter mentioned in the technical writers meeting that you would like more information added to the TLS document. Let me know what to add.

**Comment by patrickbirch** [comment]
Updated version based on discussion and meetings.

**Comment by patrickbirch** [comment]
Updated document based on Mykola conversation this morning. Mykola, please review and let me know. I also added the changes to the Slack channel.

File: source/TLS.rst [hunk 3]
```diff
@@ -0,0 +1,123 @@
+Transport Layer Security (TLS)
+******************************
+
+Security is configured at multiple levels. Transport Layer Security
+(TLS) secures the API endpoints and is a
+cryptographic protocol. TLS provides secure communications over a computer
+network.
+
+The Percona Kubernetes Operator for PXC uses a cert-manager and supports manual configuration, which is available for all versions of K8s and is an upstream feature. A cert-manager is a Kubernetes tool widely used for to automate the management and issuance of TLS certificates.
```
Comment (delgod) [inline source/TLS.rst]:
```suggestion
The Percona Kubernetes Operator for PXC can use a cert-manager for automatic certificates generation and supports also manual certificates generation. A cert-manager is a Kubernetes certificate management controller which widely used to automate the management and issuance of TLS certificates.
```

File: source/TLS.rst [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+You define a Certificate Authority certificate, which is called the root certificate. From the command line, use base64 to encode the key and certificate values. Run the following commands::
+
+  echo tls.crt | base64
+  echo tls.key | base64
+
+Copy and paste each encoded value into the appropriate sections of the YAML file. Each value is one line.
+
+You can use then use the YAML file to create the secret::
+
+  kubectl create -f secret-ssl.yaml
```
Comment (delgod) [inline source/TLS.rst]: lines 95-118 not needed, they have no sense because of command from line 91 creates secret.

File: source/TLS.rst [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
```
Comment (delgod) [inline source/TLS.rst]: it is needed to explain that newly created secret should be specified in `sslSecretName` option.
also, it is needed to explain that it is needed to generate another certificate for internal communication and put it into `sslInternalSecretName`

File: source/TLS.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
+TLS
+===
+
+Security is configured at multiple levels. Transport Layer Security
+(TLS) secures the API endpoints and is a
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/TLS.rst]: Updated to:
 Transport Layer Security (TLS) is a cryptographic protocol.

Removed mention of API endpoint.

File: source/TLS.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+cryptographic protocol. TLS provides secure communications over a computer
+network.
+
+We recommend that you run your cluster with the TLS protocol enabled. For demonstration purposes, you can disable the TLS protocol by editing cr.yaml/pxc/allowUnstafeConfigurations and changing the value to `true`. Be sure to reset the value when you have completed your tasks.
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/TLS.rst]: Removed the last sentence.

File: source/TLS.rst [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+  * Create a namespace
+  * Disable resource validations on the cert-manager namespace
+  * Install the cert-manager.
+
+The following commands perform the needed actions:
+
+::
+
+    kubectl label namespace cert-manager certmanager.k8s.io/disable-validation=true
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/TLS.rst]: Added suggested text - kubectl command to label a namespace, create namespace, label namespace - based on example provided.

File: source/TLS.rst [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    kubectl apply -f https://raw.githubusercontent.com/jetstack/cert-manager/release-0.7/deploy/manifests/cert-manager.yaml
+
+After the installation, you can verify the cert-manager by running the following command:
+
+::
+
+  kubectl get pods <cert-manager namespace>
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/TLS.rst]: Added kubectl get pods -n cert-manager to line 41

File: source/TLS.rst (hunk 3 above)
Comment (patrickbirch) [reply source/TLS.rst]: I didn't see any change of wording in the suggested change. I removed the extra "for" in the line 9.

File: source/TLS.rst (hunk 4 above)
Comment (patrickbirch) [reply source/TLS.rst]: Lines 99 - 118 are removed.

File: source/TLS.rst (hunk 5 above)
Comment (patrickbirch) [reply source/TLS.rst]: updated based on Slack conversation.

File: source/TLS.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (delgod) [reply source/TLS.rst]: currently, I see text
```
kubectl label...
kubectl create...
kubectl label...
kubectl apply...
```

but should be 
```
kubectl create...
kubectl label...
kubectl apply...
```

File: source/TLS.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (delgod) [reply source/TLS.rst]: I see 
```
kubectl get pods <cert-manager namespace>
kubectl get pods -n cert-manager
```
but should be
```
kubectl get pods -n cert-manager
```

File: source/TLS.rst (hunk 4 above)
Comment (delgod) [reply source/TLS.rst]: I still not needed text
```
You can use the YAML file to create the secret::

   kubectl create -f secret-ssl.yaml
```

File: source/TLS.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
+Transport Layer Security (TLS)
+******************************
+
+Security is configured at multiple levels. Transport Layer Security
+(TLS) is a
+cryptographic protocol. TLS provides secure communications over a computer
+network.
+
+The Percona Kubernetes Operator for PXC uses a cert-manager and supports manual configuration, which is available for all versions of Kubernetes and is an upstream feature. A cert-manager is a Kubernetes tool widely used to automate the management and issuance of TLS certificates.
```
Comment (delgod) [inline source/TLS.rst]:
```suggestion
The Percona Kubernetes Operator for PXC can use a cert-manager for automatic certificates generation and also supports manual certificates generation. A cert-manager is a Kubernetes certificate management controller which widely used to automate the management and issuance of TLS certificates.
```

File: source/TLS.rst (hunk 5 above)
Comment (delgod) [reply source/TLS.rst]: don't see any updates

File: source/TLS.rst (hunk 3 above)
Comment (delgod) [reply source/TLS.rst]: The current wording is not very precise technically.

**Pull request merged**